include/re_sa.h

```diff
@@ -59,5 +59,12 @@ bool     sa_is_linklocal(const struct sa *sa);
 bool     sa_is_loopback(const struct sa *sa);
 bool     sa_is_any(const struct sa *sa);
 
+#ifdef HAVE_INET6
+void     sa_set_scopeid(struct sa *sa, int scopeid);
+int      sa_scopeid(const struct sa *sa);
+int      sa_cpy_scopeid(struct sa *sa_dst, const struct sa *sa_src);
```

Comment:
I think this should be baked into `sa_cpy`

Reply:
It is already done via memcpy.
`sa_cpy_scopeid()` is needed to copy the scope ID of a local IPv6ll address to a destination IPv6ll address. As far as I know this selects the correct network interface. This has also be done if there is only one network interface where the scope ID is often 2 (not zero).